code-gen/src/main/scala/scalapb/zio_grpc/ZioCodeGenerator.scala

```diff
@@ -547,20 +524,35 @@ class ZioFilePrinter(
             .add(
               "// Returns a client that gives access to headers and trailers from server's response",
               s"def withResponseMetadata: ${clientWithResponseMetadataServiceName.name} = underlying",
-              "// Returns new instance with modified call options. See ClientMethods for more.",
-              s"override def mapCallOptions(f: $CallOptions => $CallOptions): ${clientServiceName.name} = new ServiceStub(underlying.mapCallOptions(f))",
-              "// Returns new instance with modified metadata. See ClientMethods for more.",
-              s"override def mapMetadataZIO(f: $SafeMetadata => zio.UIO[$SafeMetadata]): ${clientServiceName.name} = new ServiceStub(underlying.mapMetadataZIO(f))"
+              s"override def transform(t: $ZTransform[$ClientCallContext, $ClientCallContext]): ${clientServiceName.name} = new ServiceStub(underlying.transform(t))"
             )
         )
         .add("}")
         .add("")
         .add(
-          s"def scoped(managedChannel: $ZManagedChannel, options: $CallOptions = $CallOptions.DEFAULT, metadata: zio.UIO[$SafeMetadata]=$SafeMetadata.make): zio.ZIO[zio.Scope, Throwable, ${clientServiceName.name}] = ${clientWithResponseMetadataServiceName.name}.scoped(managedChannel, options, metadata).map(client => new ServiceStub(client))"
+          s"def scoped(managedChannel: $ZManagedChannel, transforms: $ZTransform[$ClientCallContext, $ClientCallContext]): zio.ZIO[zio.Scope, Throwable, ${clientServiceName.name}] ="
+        )
+        .add(
+          s"  ${clientWithResponseMetadataServiceName.name}.scoped(managedChannel, transforms).map(client => new ServiceStub(client))"
+        )
+        .add(
+          s"def scoped(managedChannel: $ZManagedChannel, options: $CallOptions = $CallOptions.DEFAULT, metadata: zio.UIO[$SafeMetadata]=$SafeMetadata.make): zio.ZIO[zio.Scope, Throwable, ${clientServiceName.name}] ="
+        )
+        .add(
+          s"  ${clientServiceName.name}.scoped(managedChannel, $ClientTransform.withCallOptions(options).andThen($ClientTransform.withMetadataZIO(metadata)))"
```

Comment:
Does your comment about unintuitive GTransform composition applies here? I think your comment says that `withMetadataZIO` would be ignored.

Reply:
In this case either way is OK (`a andThen b` vs. `a compose b`) because these two transforms act on different fields of the `CallClientContext` so the order doesn't matter.

The issue with `withOptions` and `withTimeout` is that they both modify the `CallOptions` and one can erase the change of the other if they are done in the wrong order.